src/runtime_src/core/common/scheduler.cpp

```diff
@@ -104,7 +104,12 @@ create_data_bo(xclDeviceHandle handle, size_t sz, uint32_t flags)
   ubo->dev = handle;
   ubo->bo = xclAllocBO(ubo->dev,sz,0,flags);
   ubo->data = xclMapBO(ubo->dev,ubo->bo,true /*write*/);
-  xclGetBOProperties(ubo->dev, ubo->bo, &ubo->prop);
+  auto ret = xclGetBOProperties(ubo->dev, ubo->bo, &ubo->prop);
+  if (ret < 0) {
+    xclUnmapBO(ubo->dev, ubo->bo, ubo->data);
+    xclFreeBO(ubo->dev, ubo->bo);
+    return NULL;
+  }
```

Comment:
As coded here, this will break and segmentation fault the code that uses the returned pointer.   I think it makes sense to check the return value, but it does *not* make sense to return nullptr (btw. never use NULL is C++ code).   So simply change to 
```
if (auto ret = xcGetBOProperties(...))
  throw xrt_core::error(ret,"Failed to get BO properties");
```